test/integration_test/kubevirt_hyperconv_test.go

```diff
@@ -280,6 +280,82 @@ func kubeVirtHypercVPSFixJob(t *testing.T) {
 	log.Infof("Test status at end of %s test: %s", t.Name(), testResult)
 }
 
+// This test simulates OCP upgrade by live-migrating all VMs from a node a NON-replica node and
+// then restarting PX on the node where volume is attached.
+// It expects that all the VMs from that node should end up with a bind-mount (hyperconvergence).
+// PX performs a single live migration for each VM on the node.
+// The above steps are repeated for all worker nodes in the cluster
+func kubeVirtHypercPerNodeOneLiveMigration(t *testing.T) {
+	var testrailID, testResult = 93196, testResultFail
+	runID := testrailSetupForTest(testrailID, &testResult)
+	defer updateTestRail(&testResult, testrailID, runID)
+	instanceID := "per-node"
+
+	ctxs := kubevirtVMScaledDeployAndValidate(
+		t,
+		instanceID,
+		[]string{
+			"kubevirt-fedora", "kubevirt-fedora-wait-first-consumer", "kubevirt-fedora-multi-disks-wffc",
+			"kubevirt-windows-22k-server", "kubevirt-windows-22k-server-wait-first-consumer",
+		},
+		kubevirtScale,
+	)
+	allNodes := node.GetNodesByVoDriverNodeID()
+
+	nodeVMMap := make(map[string][]*kubevirtTestState)
+	for _, currNode := range allNodes {
+		fmt.Printf("Current node: %s", currNode.Name)
+	}
+
+	//for _, appCtx := range ctxs {
+	//	vols, err := schedulerDriver.GetVolumes(appCtx)
+	//	require.NoError(t, err, "failed to get volumes for context %s", appCtx.App.Key)
+	//}
+	for _, appCtx := range ctxs {
+		testState := &kubevirtTestState{
+			appCtx:   appCtx,
+			allNodes: allNodes,
+		}
+		gatherInitialVMIInfo(t, testState)
+		verifyInitialVMI(t, testState)
+
+		// Add this testState to the node to VM map
+		vmNode := testState.vmPod.Spec.NodeName
+		nodeVMMap[vmNode] = append(nodeVMMap[vmNode], testState)
+
+	}
+	for node, testStates := range nodeVMMap {
+		log.Infof("\nProcessing Node: %s for VMs deployed", node)
+		for _, testState := range testStates {
+			log.Infof("Processing VM: %s on node: %s", testState.vmiName, node)
+			// Simulate OCP node upgrade by:
+			// 1. Live migrate the VM to another node
+			// 2. Restart PX on the original node where the volume should still be attached
+
+			// start a live migration and wait for it to finish
+			// vmPod changes after the live migration
+			startAndWaitForVMIMigration(t, testState, false /* expectReplicaNode */)
+
+			// restart px on the original node and make sure that the volume attachment has moved
+			restartVolumeDriverAndWaitForAttachmentToMove(t, testState)
```

Comment:
This does not look right. We need to do the following:
- live migrate all the VMs on this node
- restart volume driver on this node (only once)
- wait for all attachements on this node to move to another node
- do verification for all the VMs that were running on this node